examples/freertos/CMakeLists.txt

```diff
@@ -1,27 +1,169 @@
+# Cube repository because more recent kernels might be incompatible to the CMSIS
+# implementation provided by STM
+option(USE_CMSIS_RTOS "Use CMSIS RTOS provided by Cube repository" OFF)
+option(USE_CMSIS_RTOS_V2 "Use CMSIS RTOS_V2 provided by Cube repository" OFF)
+option(USE_CUBE_FREERTOS "Use the FreeRTOS kernel provided by the Cube repository" ON)
+
+if(USE_CUBE_FREERTOS)
+    message(STATUS "Using FreeRTOS provided by Cube repository")
+else()
+    message(STATUS "Using FreeRTOS from kernel repository")
+endif()
+
+if(USE_CMSIS_RTOS AND USE_CMSIS_RTOS_V2)
+    message(FATAL_ERROR "Can not use USE_CMSIS_RTOS_V2 together with USE_CMSIS_RTOS!")
+endif()
+
+if(USE_CMSIS_RTOS)
+    message(STATUS "Compiling CMSIS RTOS support")
+elseif(USE_CMSIS_RTOS_V2)
+    message(STATUS "Compiling CMSIS RTOS V2 support")
+endif()
+
+# This must come before the project call!
+set(CMAKE_TOOLCHAIN_FILE ${STM32_CMAKE_PATH}/cmake/stm32_gcc.cmake)
+project(${PROJ_NAME} CXX C ASM)
 set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
 
-find_package(CMSIS COMPONENTS STM32F407VG REQUIRED)
-find_package(HAL COMPONENTS STM32F407VG REQUIRED)
-find_package(FreeRTOS COMPONENTS ARM_CM4F REQUIRED)
+# Can be used to print out all devices for the H7 family
+# include(${STM32_CMAKE_PATH}/cmake/stm32/devices.cmake)
+# stm32_print_devices_by_family(FAMILY H7)
+# stm32_print_devices_by_family(FAMILY F4)
+
+set(SUPPORTED_BOARDS F407VG H743ZI)
+option(FREERTOS_F407VG_EXAMPLE "Compile FreeRTOS example for the F407VG board" OFF)
+option(FREERTOS_H743ZI_EXAMPLE "Compile FreeRTOS example for the H743ZI board" OFF)
+
+if(NOT FREERTOS_F407VG_EXAMPLE AND NOT FREERTOS_H743ZI_EXAMPLE)
+    message(FATAL_ERROR
+        "Please select at least one target to compile by passing FREERTOS_<BOARD>_EXAMPLE=ON\n"
+        "Supported boards: ${SUPPORTED_BOARDS}"
+    )
+endif()
+
+set(HAL_COMP_LIST RCC GPIO CORTEX)
+set(CMSIS_COMP_LIST "")
+set(FREERTOS_COMP_LIST "")
+set(FREERTOS_NAMESPACE FreeRTOS)
+
+if(USE_CMSIS_RTOS)
+    list(APPEND CMSIS_COMP_LIST RTOS)
+endif()
+if(USE_CMSIS_RTOS_V2)
+    list(APPEND CMSIS_COMP_LIST RTOS_V2)
+endif()
+
+if(FREERTOS_H743ZI_EXAMPLE)
+    list(APPEND CMSIS_COMP_LIST STM32H743ZI STM32H7_M7)
+    list(APPEND HAL_COMP_LIST STM32H7M7 STM32H743ZI)
+    list(APPEND FREERTOS_COMP_LIST ARM_CM7)
+    list(APPEND FREERTOS_COMP_LIST STM32H7)
+    if(USE_CUBE_FREERTOS)
+        set(FREERTOS_H7_NAMESPACE ${FREERTOS_NAMESPACE}::STM32::H7::M7)
+    else()
+        set(FREERTOS_H7_NAMESPACE ${FREERTOS_NAMESPACE})
+    endif()
+endif()
+
+if(FREERTOS_F407VG_EXAMPLE)
+    list(APPEND CMSIS_COMP_LIST STM32F407VG)
+    list(APPEND HAL_COMP_LIST STM32F407VG)
+    list(APPEND FREERTOS_COMP_LIST ARM_CM4F)
+    if(USE_CMSIS_RTOS OR USE_CMSIS_RTOS_V2)
+        list(APPEND FREERTOS_COMP_LIST STM32F4)
+        set(FREERTOS_F4_NAMESPACE ${FREERTOS_NAMESPACE}::STM32::F4)
+     else()
+        set(FREERTOS_F4_NAMESPACE ${FREERTOS_NAMESPACE})
+    endif()
+endif()
```

Comment:
I don't get why is there a difference in the `if` handling between the families ? Is this normal ?

Reply:
This is a possible bug, I made the if handling consistent now